helix-core/src/main/java/org/apache/helix/util/StatusUpdateUtil.java

```diff
@@ -492,47 +496,39 @@ void publishStatusUpdateRecord(ZNRecord record, Message message, Level level,
 
     Builder keyBuilder = accessor.keyBuilder();
     if (!_recordedMessages.containsKey(message.getMsgId())) {
-      if (isController) {
-        accessor
-            .updateProperty(keyBuilder.controllerTaskStatus(statusUpdateSubPath, statusUpdateKey),
-                new StatusUpdate(createMessageLogRecord(message)));
+      ZNRecord statusUpdateRecord = createMessageLogRecord(message);
+      PropertyKey propertyKey;
 
+      if (isController) {
+        propertyKey = keyBuilder.controllerTaskStatus(statusUpdateSubPath, statusUpdateKey);
+        accessor.updateProperty(propertyKey, new StatusUpdate(statusUpdateRecord));
```

Comment:
nit, we can move this line out of the if-else block too.

Reply:
fixed.